charts/emissary-ingress/templates/service.yaml

```diff
@@ -66,16 +66,25 @@ spec:
     {{- end}}
   selector:
     {{- if .Values.service.selector }}
-    {{ toYaml .Values.service.selector | nindent 6 }}
+    {{ toYaml .Values.service.selector | nindent 4 }}
     {{- else }}
     app.kubernetes.io/name: {{ include "ambassador.name" . }}
     app.kubernetes.io/instance: {{ .Release.Name }}
     {{- end }}
+    {{- if  .Values.canary}}
+    {{- if or (not (hasKey .Values.canary "mixPods")) (eq .Values.canary.mixPods false) }}
+    {{- if .Values.service.selector -}}
```

Comment:
hm i don't think i understand this. .Values.service.selector is already added on 69, and this "profile: main" bit on 77 seems to be the same thing as line 79?